clients/src/main/java/org/apache/kafka/common/requests/AddPartitionsToTxnRequest.java

```diff
@@ -101,15 +130,61 @@ public String toString() {
     public AddPartitionsToTxnRequest(final AddPartitionsToTxnRequestData data, short version) {
         super(ApiKeys.ADD_PARTITIONS_TO_TXN, version);
         this.data = data;
+        this.version = version;
     }
-
+    
+    // Only used for versions < 4
     public List<TopicPartition> partitions() {
         if (cachedPartitions != null) {
             return cachedPartitions;
         }
         cachedPartitions = Builder.getPartitions(data);
         return cachedPartitions;
     }
+    
+    private List<TopicPartition> partitionsForTransaction(String transaction) {
```

Comment:
sorry, i meant the return value `List<TopicPartition>`, it doesn't seem to be used

Reply:
?? are you saying the method isn't used? It's used on line 168

Reply:
i'm saying the method can be of type void, no?

Reply:
not as it is written. We return on line 150. I originally thought this could be useful if we just wanted the list of the partitions. but since it is private I can change.